Common/AnatupleProducer.py

```diff
@@ -28,8 +28,11 @@ def DefineAndAppend(df, varToDefine, varToCall):
     df = DefineAndAppend(df,"X_mass", f"static_cast<int>({X_mass})")
     is_data = 'true' if isData else 'false'
     df = DefineAndAppend(df,"is_data", is_data)
-    df = Baseline.RecoLeptonsSelection(df)
+    
+    df = Baseline.RecoP4(df)
+    df = DefineAndAppend(df, f"Tau_recoJetMatchIdx", f"FindMatching(Tau_p4, Jet_p4, 0.3)")
```

Comment:
- use 0.5
- do the same for electrons and muons
- reuse the matching information when filling anaTuple below (e.g. can add it to the httCand)